### PR TITLE
Adding top team API DB refresh

### PR DESCRIPTION
**Pull request by jamesmorgan**
* [x] Refresh top teams on `Events`
	* [x] Token ID changes  
	* [x] Token owner changes `to`/`from`
* [x] Create script to batch refresh top teams
* [x] Expose API
	* [x] Get top X teams
	* [x] Get top teams for address
* [x] Wire up to the frontend

### CODE REVIEW

File: functions/api/squad/index.js [hunk 1]
```diff
@@ -2,69 +2,31 @@ const _ = require('lodash');
...
+            .json(data);
+    } catch (e) {
+        next(e);
+    }
+});
 
-        const squadAverage = hasFullTeam
-            ? Math.floor(squadTotal / 11)
-            : 0;
+squad.get(`/league`, async (req, res, next) => {
```
Comment (andygray) [inline functions/api/squad/index.js]: 👍

File: functions/services/contracts/niftyFootball.contract.service.js [hunk 2]
```diff
@@ -258,6 +260,27 @@ class NiftyFootballContractService {
...
+        return owner !== ZERO_ADDRESS
+            ? owner
+            : null;
+    }
+
+    async tokenIdPointer(network = 1) {
+        console.log(`Getting token ID pointer on network [${network}]`);
+        const token = connectToToken(network);
+        const tokenIdPointer = await token.tokenIdPointer();
+        console.log(tokenIdPointer['0']);
```
Comment (andygray) [inline functions/services/contracts/niftyFootball.contract.service.js]: remove log?

File: functions/services/data/firebase.service.js [hunk 3]
```diff
@@ -17,15 +17,20 @@ module.exports = {
...
             return firestore;
         }
-        if (!admin) {
+        if (!overAdmin && !admin) {
             throw new Error('Service not setup...!');
         }
-        firestore = admin.firestore();
+
+        // When invoking from a script (see ./scripts/) we need to bootstrap firebase earlier
+        firestore = overAdmin
```
Comment (andygray) [inline functions/services/data/firebase.service.js]: good job you left a comment - was confused...

File: functions/services/data/teams.service.js [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+
+        const topSquad = [
+            ...goalkeepers,
+            ...defence,
+            ...midfield,
+            ...strikers,
+        ];
+
+        const hasFullTeam = topSquad.length === 11;
+        const hasEmptySquad = topSquad.length === 0;
```
Comment (andygray) [inline functions/services/data/teams.service.js]: can we add the `topSquad.length` to the response so on the client we can say `8/11` or 85% complete.

File: functions/services/data/teams.service.js [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+        const getPlayer = async (tokenId) => {
+            if (tokenId === 0) {
+                return {};
+            }
+            return await cardService.getCardById(network, tokenId);
+        };
+
+        // Explode out each team to be the fully populated objects
+        topTeam.team = {
+            goalkeepers: fillPosition(_.map(topTeam.team.goalkeepers, getPlayer), DEFAULT_FORMATION.goalkeepers, {}),
```
Comment (andygray) [inline functions/services/data/teams.service.js]: empty object and `0` should be one or the other?

Comment (jamesmorgan) [reply functions/services/data/teams.service.js]: fixed 👍

File: functions/services/data/teams.service.js (hunk 4 above)
Comment (jamesmorgan) [reply functions/services/data/teams.service.js]: fixed

File: functions/services/contracts/niftyFootball.contract.service.js (hunk 2 above)
Comment (jamesmorgan) [reply functions/services/contracts/niftyFootball.contract.service.js]: 👍

File: functions/services/data/teams.service.js [hunk 6]
```diff
@@ -0,0 +1,272 @@
...
+            ...worstDefence,
+            ...worstMidfield,
+            ...worstStrikers,
+        ];
+
+        const teamSize = topTeam.length;
+        const hasFullTeam = teamSize === 11;
+        const hasEmptySquad = teamSize === 0;
+
+        const teamTotal = getTeamTotal(hasFullTeam, topTeam);
```
Comment (andygray) [inline functions/services/data/teams.service.js]: should be `topTeamTotal` ?

File: functions/services/data/teams.service.js [hunk 7]
```diff
@@ -0,0 +1,272 @@
...
+
+                goalkeepers: fillPosition(_.map(worstGoalkeepers, 'tokenId'), DEFAULT_FORMATION.goalkeepers, 0),
+                defence: fillPosition(_.map(worstDefence, 'tokenId'), DEFAULT_FORMATION.defence, 0),
+                midfield: fillPosition(_.map(worstMidfield, 'tokenId'), DEFAULT_FORMATION.midfield, 0),
+                strikers: fillPosition(_.map(worstStrikers, 'tokenId'), DEFAULT_FORMATION.strikers, 0)
+            }
+        };
+
+        return firestore
+            .collection(`top-team`)
```
Comment (andygray) [inline functions/services/data/teams.service.js]: does the collection needs a name change? `teams`?

File: functions/services/data/teams.service.js [hunk 8]
```diff
@@ -0,0 +1,272 @@
...
+            .doc(getNetwork(network))
+            .collection(`address`)
+            .doc(utils.getAddress(address))
+            .set(topTeamSummary)
+            .then(() => {
+                return topTeamSummary;
+            });
+    }
+
+    async refreshTopTeamForTokenOwner(network, tokenId) {
```
Comment (andygray) [inline functions/services/data/teams.service.js]: `refreshTeamForTokenOwner`

File: package.json [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+  "name": "nifty-football-api",
+  "version": "0.0.1",
+  "description": "nifty-football-api",
+  "scripts": {},
+  "author": "BlockRocket.tech",
+  "license": "MIT",
+  "devDependencies": {
+    "commander": "^2.20.0",
+    "firebase-admin": "^7.3.0",
+    "nifty-football-contract-tools": "^1.6.0"
```
Comment (andygray) [inline package.json]: this is on `1.8.1` now if you want to bump

**Pull request merged**